File: Data Strucutres/L5-DictionaryADT-java/DictionaryTest.c
// DictionaryTest.c
// Kevin Duong
// keduong
// 12M - Lab5
// May 8th, 2016
// Test file for Dictionary.c

#include<stdio.h>
#include<stdlib.h>
#include "Dictionary.h"


int main(int arc, char* argv[]){
   char* x;
   Dictionary testStack = newDictionary();
   
   char* a = "it's"; char* b = "if"; char* c = "makes";
   char* d = "working"; char* e = "this"; char* f = "sense";
   insert(testStack, a,d);
   insert(testStack, b,e);
   insert(testStack, c,f);
   printDictionary(stdout, testStack);

   delete(testStack,b);
   printDictionary(stdout, testStack);

   x = lookup(testStack,a);
   printf("%s\n",x);
   x = lookup(testStack,b);
   printf("%s\n",x);   
   x = lookup(testStack,c);
   printf("%s\n",x);

   makeEmpty(testStack);
   printDictionary(stdout, testStack);

   freeDictionary(&testStack);
   return (EXIT_SUCCESS);
}

File: Data Strucutres/P3-DictionaryADT-java/DictionaryTest.java
// --------------------------------------
// DictionaryTest.java
// Kevin Duong
// keduong
// 12B - PA3
// April 20th, 2016
// Test for Dictionary.java
// --------------------------------------

class DictionaryTest{
   public static void main (String[] args){
      Dictionary A = new Dictionary();
    
      System.out.println(A.isEmpty());
      A.insert("a","1");
      System.out.println(A.isEmpty());
      A.insert("b","2");
      System.out.println(A.size());
    
      A.insert("c","2");
      A.insert("d","3");
      A.insert("e","4");
      A.insert("f","5");
      A.delete("g");
      A.insert("d","6"); //throws exception 
      A.toString();
   
      System.out.println(A);
      System.out.println(A.lookup("a"));
 
      A.delete("d");
      System.out.println(A.size());
      System.out.println(A.lookup("d"));

      System.out.println(A.lookup("h"));
      System.out.println(A.size());
      System.out.println(A);
      A.makeEmpty();
      System.out.println(A.size() + " " + A.isEmpty());
    
  }
}